components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -680,6 +680,12 @@ public static Notification buildNotification(Event event, Map<String, String> pl
             if (userClaims.containsKey(NotificationConstants.EmailNotification.CLAIM_URI_EMAIL)) {
                 sendTo = userClaims.get(NotificationConstants.EmailNotification.CLAIM_URI_EMAIL);
             }
+            // If OIDC dialect is used, claim URI for user email address will be "email".
+            // Hence, if sendTo is still empty, check whether "email" claim is available.
+            if (StringUtils.isEmpty(sendTo) &&
+                    userClaims.containsKey(NotificationConstants.EmailNotification.OIDC_CLAIM_URI_EMAIL)) {
+                sendTo = userClaims.get(NotificationConstants.EmailNotification.OIDC_CLAIM_URI_EMAIL);
+            }
```

Comment:
is this fix given for federated users who don't provision ?

Why only OIDC dialect's remote claim is considered?

 `event.getEventProperties().get(NotificationConstants.FEDERATED_USER_CLAIMS)` isn't this contain local claim to federated claim mappings?

Reply:
> is this fix given for federated users who don't provision ?

This fix was given for any federated user with OIDC IdPs.

> Why only OIDC dialect's remote claim is considered?

For SAML IdP federated users, once the claim mapping are added in the Console,  `event.getEventProperties().get(NotificationConstants.FEDERATED_USER_CLAIMS)` contains the email in a claim mapping with `http://wso2.org/claims/emailaddress` as the local claim URI. This is already checked at [1].

Moreover, when sending email OTP, they will be setting the `send-to` attribute by extracting them from user claims. The attribute keys that it considers are `email` (for OIDC dialect) and `http://wso2.org/claims/emailaddress` [2].

> `event.getEventProperties().get(NotificationConstants.FEDERATED_USER_CLAIMS)` isn't this contain local claim to federated claim mappings?

For an OIDC IdP (ex: Google) federated user, `event.getEventProperties().get(NotificationConstants.FEDERATED_USER_CLAIMS)` contained the email in a claim with `email` as the local claim URI. I did not contain a claim mapping with `http://wso2.org/claims/emailaddress` local claim URI.

[1] - https://github.com/wso2-extensions/identity-event-handler-notification/blob/8062ab05b49304feb58ab34adf3451ed7d1de983/components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java#L680-L682
[2] - https://github.com/wso2-extensions/identity-local-auth-emailotp/blob/9108a97014f310546205c8da50f08cede524a0cc/components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java#L1151-L1186